packages/desktop/src/protocol.rs

```diff
@@ -110,3 +106,42 @@ fn get_asset_root() -> Option<PathBuf> {
 
     None
 }
+
+/// Get the mime type from a path-like string
+fn get_mime_from_path(trimmed: &str) -> Result<&str> {
+    if trimmed.ends_with(".svg") {
+        return Ok("image/svg+xml");
+    }
+
+    let res = match infer::get_from_path(&trimmed)?.map(|f| f.mime_type()) {
+        Some(t) if t == "text/plain" => get_mime_by_ext(trimmed),
+        Some(f) => f,
+        None => get_mime_by_ext(trimmed),
+    };
+
+    Ok(res)
+}
+
+/// Get the mime type from a URI using its extension
+fn get_mime_by_ext(trimmed: &str) -> &str {
```

Comment:
Switching to infer for more accurate filetypes makes sense, but I wonder if keeping mime_guess as a backup instead of our own implementation might be better here. It supports [a lot more file types](https://github.com/abonander/mime_guess/blob/master/src/mime_types.rs) than infer, so it might be good to have as a backup.